plugin/storage/grpc/shared/grpc_server.go

```diff
@@ -78,13 +78,33 @@ func (s *grpcServer) GetServices(ctx context.Context, r *storage_v1.GetServicesR
 }
 
 // GetOperations returns the operations of a given service
-func (s *grpcServer) GetOperations(ctx context.Context, r *storage_v1.GetOperationsRequest) (*storage_v1.GetOperationsResponse, error) {
-	operations, err := s.Impl.SpanReader().GetOperations(ctx, r.Service)
+func (s *grpcServer) GetOperations(
+	ctx context.Context,
+	r *storage_v1.GetOperationsRequest,
+) (*storage_v1.GetOperationsResponse, error) {
+	operations, err := s.Impl.SpanReader().GetOperations(ctx, &spanstore.OperationQueryParameters{
+		ServiceName: r.Service,
+		SpanKind:    r.SpanKind,
+	})
 	if err != nil {
 		return nil, err
 	}
+	grpcOperation := make([]*storage_v1.Operation, len(operations))
+	operationNameSet := make(map[string]struct{})
+	for i, operation := range operations {
+		grpcOperation[i] = &storage_v1.Operation{
+			Name:     operation.Name,
+			SpanKind: operation.SpanKind,
+		}
+		operationNameSet[operation.Name] = struct{}{}
+	}
+	var uniqueNames []string
+	for name := range operationNameSet {
+		uniqueNames = append(uniqueNames, name)
+	}
 	return &storage_v1.GetOperationsResponse{
-		Operations: operations,
+		Operations:   uniqueNames,
```

Comment:
there is no need to return this field. If the plugin uses this version of grpc server pkg, it is already working with client that understands the new field.

Reply:
Are there any external grpc client that make send request to the server?

Reply:
you may want to review how plugins work: https://github.com/jaegertracing/jaeger/tree/master/plugin/storage/grpc

The client always runs in jaeger binary. But the plugin implements the server and could be from previous versions of Jaeger, so it will only return operation names.

Reply:
Oh, you mean new grpc client send request to old grpc server. That's interesting use case. In my previous work, we only have issue of old client against new server.